components/grade-table.js

```diff
@@ -1,23 +1,53 @@
 class GradeTable {
-  constructor(tableElement) {
+  constructor(tableElement, noGradesElement) {
     this.tableElement = tableElement
+    this.noGradesElement = noGradesElement
   }
   updateGrades(grades) {
     var tbody = document.querySelector('tbody')
     tbody.innerHTML = ''
 
+    var p = document.querySelector('p')
```

Comment:
Just use the properties saved in the `constructor`, instead of re-querying the whole `document`. `querySelector` is also a method of `Element`s, which only queries that `Element`'s descendants (for example, querying for a `<tbody>` inside a `<table>`).